### DISCUSSION TITLE
[SDK] Clarification on Use Cases and Necessity of 'zksync-ethers' over 'ethers'

### DISCUSSION THREAD

**Discussion by gauravl-tevaeralabs** [zkSync-Community-Hub/zksync-developers#322] [SDKs] [answered]
### Environment

Mainnet

### Acknowledgement

- [ ] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

We are transitioning to **zksync-ethers** from **zksync-web3**, and we discovered the following documentation in the NPM package for _zksync-ethers_:

`"While most of the existing SDKs should work out of the box, deploying smart contracts or utilizing unique zkSync features, like account abstraction, requires providing additional fields compared to those that Ethereum transactions have by default."`

Besides deploying contracts, do we have documentation outlining all the scenarios where using _zksync-ethers_ is necessary? Account Abstraction is mentioned, suggesting that _zksync-ethers_ might be required for paymaster functionality. Are there any other common use cases where _zksync-ethers_ is essential, and things might not function correctly with _ethers_?

We're in the process of developing an abstract SDK for our contracts that will function across any chain, primarily focusing on our layer-zero bridging functionality. As such, we plan to use _ethers_ for most cases and reserve _zksync-ethers_ for essential functionalities.

### Expected Behavior

NA

### Code Example

NA

**Comment by dutterbutter** [comment, marked as answer]
@gauravl-tevaeralabs I believe the specific functionality is captured here: https://docs.zksync.io/build/sdks/js/zksync-ethers/features.html 

@danijelTxFusion is there anything else not mentioned in that page that is relevant?

**Reply by gauravl-tevaeralabs** [reply]
Hi @dutterbutter, this is exactly what I was looking for. Thank you so much!

**Reply by danijelTxFusion** [reply]
> is there anything else not mentioned in that page that is relevant?
The page contains everything that `ethers` cannot do in the zkSync Era and `zksync-ethers` does.

**Comment by emresen30** [comment]
Thanks a lot <<<

**Comment by edevaldonunez** [comment]
The transition to zksync-ethers from zksync-web3 introduces additional fields required for deploying smart contracts and utilizing unique zkSync features like account abstraction. While most existing SDKs should work out of the box, scenarios such as deploying contracts and using account abstraction may necessitate zksync-ethers. It's crucial to consult the documentation and consider using zksync-ethers for these specific functionalities to ensure proper compatibility and functionality with zkSync.

**Comment by jodychvs** [comment]
When transitioning to zksync-ethers from zksync-web3, ensure to use it for deploying smart contracts or utilizing unique zkSync features like account abstraction. While ethers may suffice for most cases, zksync-ethers is essential for functionalities requiring zkSync-specific fields. #zkSync #zksync-ethers

**Comment by sonhoranemichelle** [comment]
The zksync-ethers SDK is essential for deploying smart contracts, utilizing unique zkSync features like account abstraction, and potentially for paymaster functionality. It's recommended for scenarios requiring interaction with zkSync-specific features beyond standard Ethereum transactions.

**Comment by mapeshewa** [comment]
The user is transitioning to zksync-ethers from zksync-web3 and is looking for documentation outlining all the scenarios where using zksync-ethers is necessary. The documentation mentions that deploying contracts or utilizing unique zkSync features, like account abstraction, requires providing additional fields compared to those that Ethereum transactions have by default. The user is developing an abstract SDK for their contracts that will function across any chain, primarily focusing on their layer-zero bridging functionality, and plans to use ethers for most cases and reserve zksync-ethers for essential functionalities.
